Create api

koskab opened the pull request.

koskab committed:
conf: added datasource and flyway configurations

koskab committed:
feat: added Item, Shop, User entities

koskab committed:
feat: supplied database with tables via .sql script

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+CREATE TABLE shops(id BIGSERIAL PRIMARY KEY,
+                   created_at TIMESTAMP,
```
нужно сделать ограничение для поле, чтобы оно не могло содержать "null"

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+CREATE TABLE shops(id BIGSERIAL PRIMARY KEY,
+                   created_at TIMESTAMP,
+                   updated_at TIMESTAMP,
+                   is_deleted SMALLINT,
```
добавить дефолтное значение 0.
Также во всех полях is_deleted сделать

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+                   email VARCHAR(30) UNIQUE,
+                   password VARCHAR(50),
+                   birthdate DATE);
+CREATE TABLE shop_items(shop_id BIGSERIAL references shops(id),
```
Вторичный ключ имеет тип данных BIGINT

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/User.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.HashMap;
+
+public class User {
+    private long id;
```
используем объект, а не примитив

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/User.java:
```diff
@@ -0,0 +1,46 @@
+    private long id;
+
+    private String name;
+    private String emailAddress;
```
По-моему тут ты забыл разделить на две переменные

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/Item.java:
```diff
@@ -0,0 +1,38 @@
+
+public class Item {
+    private long id;
+
```
+ нет полей как в таблице (created_at, updated_at и т.д.)

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/Item.java:
```diff
@@ -0,0 +1,38 @@
+    private long id;
+
+    private String name;
+    private double price;
```
Используй тип BigDecimal

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/User.java:
```diff
@@ -0,0 +1,46 @@
+
+    private String name;
+    private String emailAddress;
+    private HashMap<Item, Integer> cart;
```
Давай наверное сразу уберем эту переменную

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+                   updated_at TIMESTAMP,
+                   is_deleted SMALLINT,
+                   fullname VARCHAR(50),
+                   email VARCHAR(30) UNIQUE,
```
добавь ограничение "not null"

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+                   is_deleted SMALLINT,
+                   fullname VARCHAR(50),
+                   email VARCHAR(30) UNIQUE,
+                   password VARCHAR(50),
```
добавь ограничение "not null"

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+                   fullname VARCHAR(50),
+                   email VARCHAR(30) UNIQUE,
+                   password VARCHAR(50),
+                   birthdate DATE);
```
добавь ограничение "not null"

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131422__init_tables.sql:
```diff
@@ -0,0 +1,26 @@
+                   updated_at TIMESTAMP,
+                   is_deleted SMALLINT,
+                   name VARCHAR(30),
+                   deadline_date DATE,
```
добавь ограничение "not null"

koskab replied:
+

koskab committed:
fix: Add constraint NOT NULL for creation date
fix: Add default value equals to current time for creation date
fix: Change types of foreign keys in tables to BIGINT
fix: Set all deletion indicator equals 0 by default
fix: Add constraint NOT NULL for date of expirity of item
fix: Add constraint NOT NULL for user's email, password, birthdate

koskab committed:
fix: Entities' fields corresponds with tables' columns
fix: IDs now are of type Long

koskab committed:
fix: Items have price, names have NOT NULL constraint

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/Item.java:
```diff
@@ -0,0 +1,80 @@
+public class Item {
+    private Long id;
+
+    private Timestamp created_at;
```
В Java для типа "дата и время" используем в основном LocalDateTime + названия переменным даем в стиле camelCase, т.е. типа createdAt или updateAt

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/Item.java:
```diff
@@ -0,0 +1,80 @@
+
+    private Timestamp created_at;
+    private Timestamp updated_at;
+    private Date deadline_date;
```
В Java для типа "дата" используем в основном LocalDate

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/Item.java:
```diff
@@ -0,0 +1,80 @@
+    private Timestamp updated_at;
+    private Date deadline_date;
+
+    private Short is_deleted;
```
так как у нас может быть только 0 или 1, то можешь использовать тут Byte

koskab replied:
+

dkhvan-dev commented on AlabsTest/src/main/java/com/example/AlabsTest/Item.java:
```diff
@@ -0,0 +1,80 @@
+
+    private String name;
+
+    private Integer count;
```
Думаю здесь тоже можно Byte использовать

dkhvan-dev commented on AlabsTest/src/main/resources/db/migration/V202402131536__changes_in_tables.sql:
```diff
@@ -0,0 +1,21 @@
+ALTER TABLE shops ALTER COLUMN created_at SET NOT NULL;
+ALTER TABLE shops ALTER COLUMN created_at SET DEFAULT current_timestamp;
```
Вместо current_timestamp можно использовать now()

koskab replied:
+

koskab committed:
refactor: Change date types and variables names

koskab committed:
refactor: Change function returning date

koskab committed:
refactor: Rename variables and directory
